pkg/addonmanager/addon.go

```diff
@@ -51,12 +54,20 @@ func init() {
 type subscriptionAgent struct {
 	agentImage string
 	kubeClient kubernetes.Interface
+	agentInstallStrategy *agent.InstallStrategy
 }
 
-func NewAgent(agentImage string, kubeClient kubernetes.Interface) agent.AgentAddon {
+func NewAgent(agentImage string, kubeClient kubernetes.Interface, agentInstallAllStrategy bool) agent.AgentAddon {
+	var agentInstallStrategy *agent.InstallStrategy
+	agentInstallStrategy = nil
+	if agentInstallAllStrategy == true {
+		agentInstallStrategy = agent.InstallAllStrategy(addonDefaultInstallNamespace)
```

Comment:
Do you think it make sense to add another param to ask for the install namespace?
Personally, I only used the same namespace but I did notice someone added a param for not default namespace for clusteradm for the addon install so maybe there is a need for it.

I guess that can be done in a separate PR if it's needed.

Reply:
I think it makes sense.